src/souffle-compile.in

```diff
@@ -42,7 +44,10 @@ error() {
 #Exit on error
 set -e
 
-## set environment variables
+# set environment variables
+# directory where swig interface files are in souffle
+SWIGDIR=$(dirname $0)/../include/souffle/swig
+JAVADIR="$(printenv JAVA_HOME)"
```

Comment:
It might be better to move the JAVADIR/SWIGDIR inside the `if [ -n "$SWIGLANG" ]` block. Then, if JAVA_HOME isn't set you can exit with an informative error, or try to find the java home yourself with something like
`JAVADIR=$(java -XshowSettings:properties -version 2>&1|grep java.home|sed 's/^.*= //')`

Reply:
Fixed.

Reply:
To do

Reply:
Fixed